akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -54,43 +56,50 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
+        val replicatorMessageAdapter: ActorRef[Replicator.ReplicatorMessage] =
+          ctx.messageAdapter[Replicator.ReplicatorMessage] {
+            case changed: Replicator.Changed[_] @unchecked ⇒
+              ChangeFromReplicator(
+                changed.key.asInstanceOf[DDataKey],
+                changed.dataValue.asInstanceOf[ORMultiMap[ServiceKey[_], Entry]])
+          }
 
-    val setup = new Setup(ctx)
-    val registry = ShardedServiceRegistry(setup.settings.distributedKeyCount)
+        registry.allDdataKeys.foreach(key ⇒
+          setup.replicator ! Replicator.Subscribe(key, replicatorMessageAdapter.toUntyped)
+        )
 
-    // subscribe to changes from other nodes
-    val replicatorMessageAdapter: ActorRef[Replicator.ReplicatorMessage] =
-      ctx.messageAdapter[Replicator.ReplicatorMessage] {
-        case changed: Replicator.Changed[_] @unchecked ⇒
-          ChangeFromReplicator(
-            changed.key.asInstanceOf[DDataKey],
-            changed.dataValue.asInstanceOf[ORMultiMap[ServiceKey[_], Entry]])
-      }
+        // remove entries when members are removed
+        val clusterEventMessageAdapter: ActorRef[MemberRemoved] =
+          ctx.messageAdapter[MemberRemoved] { case MemberRemoved(member, _) ⇒ NodeRemoved(member.uniqueAddress) }
+        setup.cluster.subscribe(clusterEventMessageAdapter.toUntyped, ClusterEvent.InitialStateAsEvents, classOf[MemberRemoved])
 
-    registry.allDdataKeys.foreach(key ⇒
-      setup.replicator ! Replicator.Subscribe(key, replicatorMessageAdapter.toUntyped)
-    )
-
-    // remove entries when members are removed
-    val clusterEventMessageAdapter: ActorRef[MemberRemoved] =
-      ctx.messageAdapter[MemberRemoved] { case MemberRemoved(member, _) ⇒ NodeRemoved(member.uniqueAddress) }
-    setup.cluster.subscribe(clusterEventMessageAdapter.toUntyped, ClusterEvent.InitialStateAsEvents, classOf[MemberRemoved])
-
-    // also periodic cleanup in case removal from ORMultiMap is skipped due to concurrent update,
-    // which is possible for OR CRDTs - done with an adapter to leverage the existing NodesRemoved message
-    ctx.system.scheduler.schedule(setup.settings.pruningInterval, setup.settings.pruningInterval,
-      ctx.self.toUntyped, RemoveTick)(ctx.system.executionContext)
-
-    behavior(
-      setup,
-      registry,
-      TypedMultiMap.empty[AbstractServiceKey, SubscriptionsKV]
-    )
-  }
+        // also periodic cleanup in case removal from ORMultiMap is skipped due to concurrent update,
+        // which is possible for OR CRDTs - done with an adapter to leverage the existing NodesRemoved message
+        timers.startPeriodicTimer("remove-nodes", RemoveTick, setup.settings.pruningInterval)
+
+        // default tomstone keepalive is 24h (based on prune-gossip-tombstones-after) and keeping the actorrefs
+        // around isn't very costly so don't prune often
+        timers.startPeriodicTimer("prune-tombstones", PruneTombstonesTick, 1.hour)
```

Comment:
I kind of like using the `PruneTombstonesTick` as the timer key when it's `case object` ticks